snap/snapcraft.yaml

```
@@ -34,6 +41,7 @@ apps:
       REGISTRY_ARG: "--registry"
       DEVICE_PROFILESDIR: "$SNAP_DATA/config/device-camera/res/profiles"
       DEVICE_DEVICESDIR: "$SNAP_DATA/config/device-camera/res/devices"
+      SECRETSTORE_TOKENFILE: $SNAP_DATA/device-camera/secrets-token.json
     daemon: simple
```

Add dependency for `security-consul-bootstrapper` in `after` section (using slot?   don't know how you done in this snap but needs that dependency. )

```suggestion
    daemon: simple
    after:
       # This generates the consul role for this service before the service starts      
       - security-consul-bootstrapper
```

---

@jim-wang-intel unfortunately snapd does not support cross-snap service dependencies, so this is not possible. That said, device services have a configurable retry mechanism on startup which is used to ensure that it's dependencies are available. Since the main edgexfoundry snap always gates startup of the core-*, support-*, and SMA to be after security-consul-bootstrapper and security-proxy-setup have run, relaying on the SDK startup logic just works. If this is found to a problem (i.e. on a hw constrained device), the startup environment variables can be modified via the snap config hook.

---

ok, sounds good, thanks @tonyespy for explaining this.

---

ok so I am seeing this dependency after as resolved . but we do need to introduce that extra env. `ADD_REGISTRY_ACL_ROLES = device-camera`   into `security-consul-bootstrapper` in main snap or at least in the documentation above to state that.

---

@jim-wang-intel - thanks for raising that point. Once we have these device PRs merged we'll also raise a PR for the main edgexfoundry / edgex-go snap where we'll set both ADD_SECRETSTORE_TOKENS and ADD_REGISTRY_ACL_ROLES

---

sounds good, @siggiskulason  thanks!